driver/outer_driver.hpp

```diff
@@ -0,0 +1,517 @@
+
+    int GetandSetData() override;
+    std::vector<int> GetInputTensorLengthsFromCmdLine();
+
+    int AllocateBuffersAndCopy() override;
+
+    int RunForwardGPU() override;
+    int RunForwardCPU();
+
+    int RunBackwardGPU() override;
+    int RunBackwardCPU();
+
+    Tref GetTolerance();
+    int VerifyBackward() override;
+    int VerifyForward() override;
+    ~OuterDriver() override
+    {
+        miopenDestroyTensorDescriptor(input1Desc);
+        miopenDestroyTensorDescriptor(input1Desc);
+        miopenDestroyTensorDescriptor(yDesc);
+    }
+
+private:
+    InputFlags inflags;
+
+    int forw;
+
+    miopenTensorDescriptor_t input1Desc;
+    miopenTensorDescriptor_t input2Desc;
+    miopenTensorDescriptor_t input1GradDesc;
+    miopenTensorDescriptor_t input2GradDesc;
+    miopenTensorDescriptor_t yDesc;
+    miopenTensorDescriptor_t yGradDesc;
+
+    std::unique_ptr<GPUMem> in1_dev;
+    std::unique_ptr<GPUMem> in2_dev;
+    std::unique_ptr<GPUMem> in1Grad_dev;
+    std::unique_ptr<GPUMem> in2Grad_dev;
+    std::unique_ptr<GPUMem> out_dev;
+    std::unique_ptr<GPUMem> outGrad_dev;
+
+    std::vector<Tgpu> in1;
+    std::vector<Tgpu> in2;
+    std::vector<Tgpu> in1Grad;
+    std::vector<Tgpu> in2Grad;
+    std::vector<Tgpu> out;
+    std::vector<Tgpu> outGrad;
+
+    std::vector<Tref> in1Gradhost;
+    std::vector<Tref> in2Gradhost;
+    std::vector<Tref> outhost;
+
+    size_t ws_sizeInBytes;
+};
+
+template <typename Tgpu, typename Tref>
+int OuterDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int OuterDriver<Tgpu, Tref>::GetandSetData()
+{
+    std::vector<int> in_lens = GetInputTensorLengthsFromCmdLine();
+
+    int in_n = inflags.GetValueInt("in_n");
+    int in_m = inflags.GetValueInt("in_m");
+
+    auto lens1 = std::vector<int>({in_lens[0]});
+    auto lens2 = std::vector<int>({in_lens[1]});
+
+    SetTensorNd(input1Desc, lens1, data_type);
+    SetTensorNd(input2Desc, lens2, data_type);
+
+    SetTensorNd(input1GradDesc, lens1, data_type);
+    SetTensorNd(input2GradDesc, lens2, data_type);
+
+    std::vector<int> out_len({in_n, in_m});
+
+    SetTensorNd(yDesc, out_len, data_type);
+    SetTensorNd(yGradDesc, out_len, data_type);
+
+    return 0;
+}
+
+template <typename Tgpu, typename Tref>
+int OuterDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw", 'F', "1", "Run only Forward Sum (Default=1)", "int");
+    inflags.AddInputFlag("in_n", 'N', "128", "n size (Default=32)", "int");
+    inflags.AddInputFlag("in_m", 'M', "256", "m size(Default=32)", "int");
+
+    inflags.AddInputFlag(
+        "wall", 'w', "0", "Wall-clock Time Each Layer, Requires time == 1 (Default=0)", "int");
+    inflags.AddInputFlag("iter", 'i', "5", "Number of Iterations (Default=10)", "int");
```

Comment:
Wrong default value (or help text)

Reply:
I made the default value consistent with the message.